src/core/server/elasticsearch/version_check/ensure_es_version.test.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import _ from 'lodash';
+
+import { ensureEsVersionMatch, blockUntilEsVersionMatches, NodesInfo } from './ensure_es_version';
+import { loggingServiceMock } from '../../logging/logging_service.mock';
+
+const mockLoggerFactory = loggingServiceMock.create();
+const mockLogger = mockLoggerFactory.get('mock logger');
+
+describe('ensureVersionMatch', () => {
+  const KIBANA_VERSION = '5.1.0';
+
+  function createNodes(...args: string[]): NodesInfo {
+    const versions = _.shuffle(args);
```

Comment:
Question: Do we really like that kind of randomness in tests?

Reply:
I agree, this was in the original tests, but I've removed it from the NP tests.